dnf5/cmdline_aliases.cpp

```diff
@@ -0,0 +1,640 @@
+                                        group_id,
+                                        e.what(),
+                                        config_file_path.native(),
+                                        location.line(),
+                                        location.line_str());
+                                    logger->error("{}", msg);
+                                    std::cerr << msg << std::endl;
+                                    continue;
+                                }
+                            } else if (key == "complete") {
+                                complete = value.as_boolean();
+                            } else {
+                                auto location = value.location();
+                                logger->warning(
+                                    "Unknown attribute \"{}\" of named argument \"{}\" in file \"{}\" on line {}: {}",
+                                    key,
+                                    element_id_path,
+                                    config_file_path.native(),
+                                    location.line(),
+                                    location.line_str());
+                            }
+                        }
+
+                        if (!source) {
+                            auto msg = fmt::format(
+                                "Mising attribute \"source\" for named argument \"{}\" in file \"{}\"",
+                                element_id_path,
+                                config_file_path.native());
+                            logger->error("{}", msg);
+                            std::cerr << msg << std::endl;
+                            continue;
+                        }
+
+                        if ((!long_name || long_name->empty()) && (!short_name || short_name == '\0')) {
+                            auto msg = fmt::format(
+                                "At least one of the attributes \"long_name\" and \"short_name\" must be set for named "
+                                "argument \"{}\" in file \"{}\"",
+                                element_id_path,
+                                config_file_path.native());
+                            logger->error("{}", msg);
+                            std::cerr << msg << std::endl;
+                            continue;
+                        }
+
+                        auto alias_arg = source->add_alias(
+                            element_id, long_name ? *long_name : "", short_name ? *short_name : '\0', nullptr);
+                        alias_arg->set_complete(complete);
+
+                        if (group) {
+                            group->register_argument(alias_arg);
+                        }
+                        element_parent_cmd->register_named_arg(alias_arg);
+                        break;
+                    }
+
+                    // Creates a new named argument
+                    case ElementType::NAMED_ARG: {
```

Comment:
Maybe I have missed something but how is the `NAMED_ARG` used?
I don't see how it can connect to any functionality.

Is there a plan to have all commands defined entirely in the config so that users can use completely custom naming, descriptions etc..? and dnf5 will just hook into some expected ids?

Reply:
The `cloned_named_args` is a simplified version of the named argument alias. It basically copies the existing argument. The user defines which argument (source), where and new names (short, long). And the group and whether to offer bash completions.

The `named_arg` is more powerful. Defines a new named argument that can represent a combination of existing named arguments (defined using the `attached_named_args` table) and pass a different value to them.
So far the replaced aliases were simple and therefore it was enough to use 'cloned_named_args'.

Reply:
I see, I confused this together with the COMMAND.
Thanks for explanation.